src/Contract/Operation/Allable.php

```diff
@@ -20,7 +20,10 @@ interface Allable
      *
      * @see https://loophp-collection.readthedocs.io/en/stable/pages/api.html#all
      *
-     * @return array<TKey, T>
+     * @param bool $normalize
+     *
+     * @return list<T>|array<TKey, T>
+     * @psalm-return ($normalize is true ? list<T> : array<TKey, T>)
```

Comment:
Psalm [conditional types](https://psalm.dev/docs/annotating_code/type_syntax/conditional_types/) allow for more accurately describing the return type here

Reply:
So sad we don't have a generic way to do that :(

Besides that, it's fine.